### PR TITLE
Dynamic group link for more than 2 user fixed

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

- Dynamic group booking only supports 2 users currently because of the use of replace added in PR #2414. This PR adds support for more than 2 users by building on that PR logic.
- This PR also fixes the dynamic group link logic which earlier only allowed 2 users, but now provides the expected link with all the users as part of the booking.
- This PR updates the availability and booking pages to use Name instead of usernames of the dynamic group for title

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [x] Try booking `/pro+rick+free+usa` in staging

### CODE REVIEW

File: apps/web/pages/[user]/[type].tsx [hunk 1]
```diff
@@ -261,9 +261,15 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
...
+  const dynamicNames = isDynamicGroup
+    ? users.map((user) => {
+        return user.name || "";
+      })
+    : [];
+
   const profile = isDynamicGroup
     ? {
-        name: getGroupName(usernameList),
+        name: getGroupName(dynamicNames),
```
Comment (alishaz-polymath) [inline apps/web/pages/[user]/[type].tsx:272]: Adds user's Names instead of username for title

File: apps/web/pages/[user]/book.tsx [hunk 2]
```diff
@@ -168,9 +168,15 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
...
+  const dynamicNames = isDynamicGroupBooking
+    ? users.map((user) => {
+        return user.name || "";
+      })
+    : [];
+
   const profile = isDynamicGroupBooking
     ? {
-        name: getGroupName(usernameList),
+        name: getGroupName(dynamicNames),
```
Comment (alishaz-polymath) [inline apps/web/pages/[user]/book.tsx:179]: Uses Name isntead of username for title

File: packages/lib/defaultEvents.ts [hunk 3]
```diff
@@ -126,11 +126,10 @@ export const getGroupName = (usernameList: string[]): string => {
 export const getUsernameSlugLink = ({ users, slug }: UsernameSlugLinkProps): string => {
   let slugLink = ``;
   if (users.length > 1) {
-    let combinedUsername = ``;
-    for (let i = 0; i < users.length - 1; i++) {
-      combinedUsername = `${users[i].username}+`;
+    let combinedUsername = `${users[0].username}`;
```
Comment (alishaz-polymath) [inline packages/lib/defaultEvents.ts:129]: Fixes multi user username slug logic

**Review by hariombalhara — APPROVED** [review]
Code LGTM.

**Review by pumfleet — APPROVED** [review]

**Pull request merged**